app/controllers/RemoveItemController.scala

```diff
@@ -0,0 +1,78 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import forms.AddMoreInformationFormProvider
+import models.NormalMode
+import navigation.Navigator
+import pages.unsatisfactory.individualItems.RemoveItemPage
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.UserAnswersService
+import utils.JsonOptionFormatter
+import views.html.AddMoreInformationView
+
+import javax.inject.Inject
+import scala.concurrent.Future
+
+class RemoveItemController @Inject()(
+                                              override val messagesApi: MessagesApi,
+                                              override val userAnswersService: UserAnswersService,
+                                              override val navigator: Navigator,
+                                              override val auth: AuthAction,
+                                              override val withMovement: MovementAction,
+                                              override val getData: DataRetrievalAction,
+                                              override val requireData: DataRequiredAction,
+                                              formProvider: AddMoreInformationFormProvider,
+                                              val controllerComponents: MessagesControllerComponents,
+                                              view: AddMoreInformationView
+                                            ) extends BaseNavigationController with AuthActionHelper with JsonOptionFormatter {
+
+  def onPageLoad(ern: String,
+                         arc: String,
+                         idx: Int): Action[AnyContent] =
+    authorisedDataRequest(ern, arc) { implicit request =>
+      withItem(idx) {
+        _ => Ok(view(
+          form = fillForm(RemoveItemPage(idx), formProvider(RemoveItemPage(idx))),
+          page = RemoveItemPage(idx),
+          action = routes.RemoveItemController.onSubmit(ern, arc, idx)
+        ))
+      }
+    }
+
+  def onSubmit(ern: String,
+                       arc: String,
+                       idx: Int): Action[AnyContent] =
+    authorisedDataRequestAsync(ern, arc) { implicit request =>
+      withItemAsync(idx) {
+        _ =>
+        formProvider(RemoveItemPage(idx)).bindFromRequest().fold(
+          formWithErrors =>
+            Future.successful(BadRequest(view(formWithErrors, RemoveItemPage(idx), routes.RemoveItemController.onSubmit(ern, arc, idx)))),
+          {
+            case true =>
+              removeItemAndRedirect(idx)
+            case false =>
+              saveAndRedirect(RemoveItemPage(idx), false, NormalMode)
```

Comment:
It probably doesn't need to save this answer. As it's only used for routing purposes, also, it might be weird if they added the item again and then removed it as the previous answer would be pre-populated.

I guess there's two options:

- Don't fill the form on the `onPageLoad`
- Have the Redirect logic take place within the controller rather than being delegated to the Navigator

Reply:
I don't mind which solution we go with - I agree pre-populating "No" might seem weird for users.